packages/openfermion/quri_parts/openfermion/utils/add_parametric_pauli_rotation.py

```diff
@@ -0,0 +1,64 @@
+from typing import Sequence, Union, cast
+
+from openfermion.ops import FermionOperator
+
+from quri_parts.chem.utils.excitations import DoubleExcitation, SingleExcitation
+from quri_parts.circuit import LinearMappedUnboundParametricQuantumCircuit, Parameter
+
+from ..transforms import OpenFermionQubitOperatorMapper
+
+
+def add_parametric_pauli_rotation_gate(
+    circuit: LinearMappedUnboundParametricQuantumCircuit,
+    excitation_indices: Sequence[Union[SingleExcitation, DoubleExcitation]],
+    params: Sequence[Parameter],
+    operator_mapper: OpenFermionQubitOperatorMapper,
+    trotter_number: int,
+) -> LinearMappedUnboundParametricQuantumCircuit:
+    """Add parametric Pauli rotation gates to the given :attr:`circuit`."""
+    for i, sorb_indices in enumerate(excitation_indices):
+        op = _create_operator(sorb_indices, operator_mapper)
+        for pauli, coeff in op.items():
+            pauli_index_list, pauli_id_list = zip(*pauli)
+            coeff = coeff.imag
+            circuit.add_ParametricPauliRotation_gate(
+                pauli_index_list,
+                pauli_id_list,
+                {params[i]: -2.0 * coeff / trotter_number},
+            )
+    return circuit
```

Comment:
I prefer a name reflecting the meaning of this function: this function adds gates for specified excitations with Trotter decomposition.
And as for `trotter_number` argument, I think it is *not* directly related to Trotter decomposition performed in this function, so its name should probably be just `coef` or something.